### PR TITLE
Maker: Prefer Circuit Playground over other serial port devices

### PR DESCRIPTION
**Pull request by islemaster**
@alicesteinglass brought me a teacher's laptop today that was failing to connect to the Circuit Playground board even though all the setup instructions had been followed correctly.  This change makes that laptop (and probably many like it) [connect successfully](https://youtu.be/7Irms0j0qEA).

The first odd thing I noticed on this computer was that the maker setup page reported success for the "Board plugged in" step before I plugged the board in.  Turns out it's running something called [Intel Active Management Technology](http://www.intel.com/content/www/us/en/architecture-and-technology/intel-active-management-technology.html) which uses a "Serial-Over-Lan (SOL) driver." That driver shows up in the serialport list on the machine, and our site was trying to connect to it, thinking it was the Adafruit board.

multiple_serialports

You can see here that the Adafruit Circuit Playground board is also listed, it's just listed _after_ the SOL port which is why we were missing it. At the bottom of that screenshot you can see the `op: "construct", path: "COM3"` call that shows we were selecting the wrong port.

The solution is to teach Maker Toolkit to be much smarter about which of the available ports it connects to.  Adafruit boards (and most USB devices) report a vendor id (VID) and a product id (PID) to help identify the connected device.  We now read these ids on the available ports and pick the best one to connect to, searching in this order.

1. Look for a port that reports the Adafruit Circuit Playground VID and PID (this is going to be the majority case in schools).
2. If a Circuit Playground isn't found, look for another device reporting the Adafruit VID - it might be compatible.
3. If no Adafruit device is found, try a device that as a serialport-like port name and has any valid (nonzero) VID and PID.

These rules should catch not just this case, but probably a lot of cases we don't know about yet where some other software or hardware on the machine would have looked vaguely connectable.

### CODE REVIEW

File: apps/src/lib/kits/maker/BoardController.js [hunk 1]
```diff
@@ -16,9 +16,23 @@ import {BOARD_EVENT_ALIASES} from './PlaygroundConstants';
...
+ */
+
 /** @const {string} */
 const CHROME_APP_ID = 'ncmmhcpckfejllekofcacodljhdhibkg';
 
+/** @const {string} Adafruit's vendor id as reported by Circuit Playground boards */
+export const ADAFRUIT_VID = '0x239a';
+
+/** @const {string} The Circuit Playground product id as reported by Circuit playground boards */
+export const CIRCUIT_PLAYGROUND_PID = '0x8011';
```
Comment (Hamms) [inline apps/src/lib/kits/maker/BoardController.js:34]: how stable are these? If they were to change, do we have a good way of finding out that they did so?

File: apps/test/unit/lib/kits/maker/sampleSerialPorts.js [hunk 2]
```diff
@@ -1,39 +1,28 @@
...
-    "locationId": "",
-    "vendorId": "0x0",
-    "productId": "0x0"
+    comName: "COM5",
+    manufacturer: "Adafruit Circuit Playground",
+    serialNumber: "",
+    pnpId: "",
+    locationId: "",
+    vendorId: "0x239a",
+    productId: "0x8011"
```
Comment (Hamms) [inline apps/test/unit/lib/kits/maker/sampleSerialPorts.js:15]: is there a reason these aren't using the constants from `BoardController.js`?

File: apps/src/lib/kits/maker/BoardController.js (hunk 1 above)
Comment (islemaster) [reply apps/src/lib/kits/maker/BoardController.js:34]: I gather ([here](http://www.voti.nl/docs/usb-pid.html), [here](https://learn.adafruit.com/hacking-the-kinect/verify-the-vid-and-pid), [here](https://learn.adafruit.com/introducing-trinket/starting-the-bootloader), [here](https://learn.adafruit.com/simple-arduino-based-usb-vid-and-pid-tester/lets-do-this-thing)) that these IDs are broadly considered stable and are the standard way to identify a USB device and look up drivers. It sounds like the Adafruit vendor id was purchased/reserved by them from the [USB developers group](http://www.usb.org/developers) and that they guarantee product id to be unique within their vendor id.

These IDs are kept [in this configuration file in this repo](https://github.com/adafruit/Adafruit_Arduino_Boards/blob/master/boards.txt) to make them available to the Arduino IDE, and while I'm not sure this is good evidence there is _very_ low churn on that file.

More generally, I think my next change should introduce some metrics around connected boards.  I want to have the maker setup page survey automatically include information about the chosen serialport in cases where we report "Board Plugged In" but not "Board Connectable" to catch conflicts like this in the future, and it'd be interesting to anonymously report port info for successfully connected boards out to GA or someplace.

File: apps/test/unit/lib/kits/maker/sampleSerialPorts.js (hunk 2 above)
Comment (islemaster) [reply apps/test/unit/lib/kits/maker/sampleSerialPorts.js:15]: I was torn about this; on the one hand, this file is sort of a clean, dependencyless fixture and I didn't want to mess with that.  In fact, I wonder if this should somehow be converted to json to show that it's purely data.  On the other hand, DRY...

**Pull request merged**